Open-Rival/src/Building.cpp

```diff
@@ -0,0 +1,70 @@
+#include "pch.h"
+#include "Building.h"
+
+namespace Rival {
+
+	Building::Building(BuildingType type) :
+            deleted(false),
+            id(-1),
+            type(type),
+            animation(type),
+            wallVariant(WallVariant::A) {
+	}
+
+    void Building::addedToWorld(
+        int newId, int newPlayer, int newX, int newY, WallVariant newWallVariant) {
+        id = newId;
+        player = newPlayer;
+        x = newX;
+        y = newY;
+        wallVariant = newWallVariant;
+    }
+
+    int Building::getCurrentSpriteIndex() const {
+        auto spriteIndex = animation.getCurrentSpriteIndex();
+        bool isAWall = (type == BuildingType::TreeWall);
+        isAWall = isAWall || (type == BuildingType::GreenskinWall);
+        isAWall = isAWall || (type == BuildingType::Wall);
```

Comment:
```suggestion
        bool isAWall = (type == BuildingType::TreeWall)
                || (type == BuildingType::GreenskinWall)
                || (type == BuildingType::Wall);
```

Alternatively, we could extract this to separate method. We may need it elsewhere.

Reply:
Done. We can extract into another method when we need that.